lib/RecommendationProvider.js

```diff
@@ -0,0 +1,157 @@
+/* globals exports, require, Task */
+"use strict";
+const {Cu} = require("chrome");
+const {setTimeout, clearTimeout} = require("sdk/timers");
+const simplePrefs = require("sdk/simple-prefs");
+
+const MILLISECONDS_IN_A_MINUTE = 1000 * 60;
+const MILLISECONDS_IN_TEN_MINUTES = MILLISECONDS_IN_A_MINUTE * 10;
+const MILLISECONDS_IN_AN_HOUR = MILLISECONDS_IN_A_MINUTE * 60;
+const MILLISECONDS_IN_A_DAY = MILLISECONDS_IN_AN_HOUR * 24;
+const MILLISECONDS_IN_A_WEEK = MILLISECONDS_IN_A_DAY * 7;
+const POCKET_PREF = "pocket.endpoint";
+
+const DEFAULT_TIMEOUTS = {
+  pocketTimeout: MILLISECONDS_IN_TEN_MINUTES, // every 10 minutes, refresh the Pocket recommendations
+};
+
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.importGlobalProperties(["fetch"]);
+
+function RecommendationProvider(previewProvider, options = {}) {
+  this.options = Object.assign({}, DEFAULT_TIMEOUTS, options);
+  this._recommendedContent = [];
+  this._blockedRecommendedContent = [];
+  this._currentRecommendation = null;
+  this._pocketEndpoint = simplePrefs.prefs[POCKET_PREF];
+  this._previewProvider = previewProvider;
+  this._asyncExpireRecommendations(this.options.pocketTimeout);
+}
+
+RecommendationProvider.prototype = {
+  _pocketTimeoutID: null,
+
+  /**
+    * If we need a recommendation, go get one
+    */
+  getARecommendation(getRecommendation) {
```

Comment:
https://github.com/mozilla/activity-stream/pull/906/files#diff-cd9b1d133214f36f1c20099d7f6817e6R22 

The PR for the entire thing hooked up (including content) is here https://github.com/mozilla/activity-stream/pull/906/files